perceval/algorithm/tomography/tomography_mle.py

```diff
@@ -0,0 +1,400 @@
+        pass
+
+    @abstractmethod
+    def _povm_data(self):
+        pass
+
+    def _collect_data(self, prep_state_indices, state_meas_indices) -> list:
+        # performs measurements on the output_state for given preparation and measurement state indices at each qubit
+
+        output_distribution, self._gate_logical_perf = _compute_probs(self, prep_state_indices, state_meas_indices,
+                                                                      denormalize=False)
+
+        B = []
+        for j in range(2**self._nqubit):
+            measurement_state = BasicState()
+            for m in range(self._nqubit - 1, -1, -1):
+                if (j // (2 ** m)) % 2 == 0:
+                    measurement_state *= self._LOGICAL0
+                else:
+                    measurement_state *= self._LOGICAL1
+
+            B.append(output_distribution[measurement_state] / (3 ** self._nqubit))
+        return B
+
+    def _povm_state(self) -> list:
+        # Gives a set of measurement states, so they form a set of informationally complete measurements.
+        # For 1 qubit, they are (order important) : |0>,|1>,|+>,|->,|i+>,|i->,
+        # These measurement states are eigenvectors of the tensor products of Pauli operators
+
+        d = 2 ** self._nqubit
+        P = []
+        B = [np.array([[1], [0]], dtype='complex_'), np.array([[0], [1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [-1]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [1j]], dtype='complex_'),
+             (1 / np.sqrt(2)) * np.array([[1], [-1j]], dtype='complex_')]  #
+
+        for pauli_index in range(3 ** self._nqubit):
+            # iterates over pauli operators I(equivalent to Z),X,Y or their tensor products
+            pauli_list = []  # saves on each element the pauli index for this qubit
+
+            for j in range(self._nqubit - 1, -1, -1):
+                pauli_list.append(pauli_index // (3 ** j))
+                pauli_index = pauli_index % (3 ** j)
+            X = [np.array([1], dtype='complex_')] * d  # neutral elements for tensor products
+
+            for qubit_index in range(self._nqubit):
+                for eigenvector_index in range(d):  # iterates over the eigenvectors of the pauli operator
+                    eigenvector_list = []  # saves on each element the eigenvector index for this qubit
+                    k1 = eigenvector_index
+                    for j1 in range(self._nqubit - 1, -1, -1):
+                        eigenvector_list.append(k1 // (2 ** j1))
+                        k1 = k1 % (2 ** j1)
+                    X[eigenvector_index] = np.kron(X[eigenvector_index],
+                                                   B[2 * pauli_list[qubit_index] + eigenvector_list[qubit_index]])
+            P += X
+        return P
+
+    def _povm_operator(self) -> list:
```

Comment:
as far as I understand _povm_operator and _povm_state never change ? So why not calculate them once at the begging of this class and use it each time you need it after ?